data/abilities.js

```diff
@@ -1032,9 +1032,22 @@ let BattleAbilities = {
 		},
 		onAllySetStatus: function (status, target, source, effect) {
 			if (target.hasType('Grass')) {
-				if (!effect || !effect.status) return false;
```

Comment:
What was wrong with this style of condition to check whether the ability visibly activates?

Reply:
To fit the edge case of  allowing yawn to cause sleep if it was given before flower veil was active

Reply:
I don't have an issue with your change to line 1034; I was quoting the old line 1035 which you deleted.

Reply:
Ah, right, I'd forgotten that Safeguard was doing it differently to everywhere else. I'd obviously [meant](https://github.com/urkerab/Pokemon-Showdown/blob/b1738310c94b79ec608e8505b6a5c04e619987f9/data/moves.js#L13931-L13933) to update Safeguard at some point but I've just been too busy...